processor/k8sprocessor/kube/client.go

```diff
@@ -235,6 +236,19 @@ func (c *WatchClient) extractPodAttributes(pod *api_v1.Pod) map[string]string {
 			tags[r.Name] = c.extractField(v, r)
 		}
 	}
+
+	if c.Rules.ContainerName {
+		if len(pod.Spec.Containers) > 0 {
+			var names []string
+			for _, container := range pod.Spec.Containers {
+				names = append(names, container.Name)
+			}
+
+			sort.Strings(names)
+			tags[conventions.AttributeK8sContainer] = strings.Join(names, ",")
```

Comment:
k8s.container.name should be about a single container. I assume we can't do that here because we correlate based on IP and the IP is shared across all the containers in the pod. If we make this an array (seems better) I think we should also consider adding a new semantic convention of `k8s.pod.containers` for this purpose.

Reply:
I looked around semantic conventions to see if there's any other attribute troubled by a similar property (usually being a single value, but sometimes being a collection). The closest I get is [process.command_line](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/resource/semantic_conventions/process.md#process), but there are really no other examples.

The same problem will apply to Kubernetes Service - there might be several of them pointing to a single pod. 

I am not sure if this ambiguity could be solved in any way currently (@dashpole, perhaps you might have some suggestions?). 

This leads to several specification-related questions, such as:
- should there bet two attributes (one singular, one plural) depending if there's (typically) one or (sometimes) multiple values or, only one of them?
- if there's only one attribute name, should it be plural or singular when there might be a collection of values for a key?
- if there's only one attribute name, what should be the type of attribute value (always an array or perhaps a string when there's only single entry)?

For reference, there's a an ongoing discussion on [pluralization guidelines for metrics](https://github.com/open-telemetry/opentelemetry-specification/pull/1109)

Reply:
If this is a metric about a pod (e.g. is correlated with the Pod's IP), then IMO it shouldn't have a container attribute.  If it is about a container, then it should have a single container name.

Reply:
@dashpole so essentially when a single container cannot be accurately assigned then the information should be not included, right?

Reply:
Yes, at least in my opinion.

From monitoring systems i'm familiar with, I haven't seen a good way to handle lists of items in label (attribute) values.

The other thing to consider is that technically containers can be added to a running pod with the [debug container](https://kubernetes.io/docs/tasks/debug-application-cluster/debug-running-pod/#ephemeral-container) feature, so this attribute wouldn't necessarily be constant over the lifetime of a pod.

Reply:
> The other thing to consider is that technically containers can be added to a running pod with the debug container feature, so this attribute wouldn't necessarily be constant over the lifetime of a pod.

Even though this seems like a niche use case I agree that it might introduce incorrect data when we'd tag as above.

> From monitoring systems i'm familiar with, I haven't seen a good way to handle lists of items in label (attribute) values.

As for this I was trying to approach tags rewrite to not be of [`map[string]string` type](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/0848767bd9945705929580102541efc57d0614a9/processor/k8sprocessor/kube/kube.go#L65) but rather a map of `string` to [`AnyValue`](https://github.com/open-telemetry/opentelemetry-proto/blob/69d265b577141940067f9355d32934aa602fdcff/opentelemetry/proto/common/v1/common.proto#L24-L38) to have the tags already in the type that they will be serialized to.

This could make it easier as having a plural tag as key would already indicate there's an array in the value.